packages/firebase_auth/firebase_auth_platform_interface/lib/src/method_channel_firebase_auth.dart

```diff
@@ -0,0 +1,459 @@
+  }
+
+  @override
+  Future<void> delete(String app) {
+    return channel.invokeMethod<void>('delete', <String, String>{'app': app});
+  }
+
+  @override
+  Future<PlatformAuthResult> signInWithCredential(
+    String app,
+    AuthCredential credential,
+  ) async {
+    final Map<String, dynamic> data =
+        await channel.invokeMapMethod<String, dynamic>(
+      'signInWithCredential',
+      <String, dynamic>{
+        'app': app,
+        'provider': credential.providerId,
+        'data': credential.asMap(),
+      },
+    );
+    final PlatformAuthResult authResult = _decodeAuthResult(data);
+    return authResult;
+  }
+
+  @override
+  Future<PlatformAuthResult> signInWithCustomToken(
+    String app,
+    String token,
+  ) async {
+    final Map<String, dynamic> data =
+        await channel.invokeMapMethod<String, dynamic>(
+      'signInWithCustomToken',
+      <String, String>{'token': token, 'app': app},
+    );
+    final PlatformAuthResult authResult = _decodeAuthResult(data);
+    return authResult;
+  }
+
+  @override
+  Future<void> signOut(String app) {
+    return channel.invokeMethod<void>('signOut', <String, String>{'app': app});
+  }
+
+  @override
+  Future<PlatformIdTokenResult> getIdToken(String app, bool refresh) async {
+    final Map<String, dynamic> data = await channel
+        .invokeMapMethod<String, dynamic>('getIdToken', <String, dynamic>{
+      'refresh': refresh,
+      'app': app,
+    });
+
+    return _decodeIdTokenResult(data);
+  }
+
+  @override
+  Future<PlatformAuthResult> reauthenticateWithCredential(
+    String app,
+    AuthCredential credential,
+  ) async {
+    final Map<String, dynamic> data =
+        await channel.invokeMapMethod<String, dynamic>(
+      'reauthenticateWithCredential',
+      <String, dynamic>{
+        'app': app,
+        'provider': credential.providerId,
+        'data': credential.asMap(),
+      },
+    );
+    return _decodeAuthResult(data);
+  }
+
+  @override
+  Future<PlatformAuthResult> linkWithCredential(
+    String app,
+    AuthCredential credential,
+  ) async {
+    final Map<String, dynamic> data =
+        await channel.invokeMapMethod<String, dynamic>(
+      'linkWithCredential',
+      <String, dynamic>{
+        'app': app,
+        'provider': credential.providerId,
+        'data': credential.asMap(),
+      },
+    );
+    final PlatformAuthResult result = _decodeAuthResult(data);
+    return result;
+  }
+
+  @override
+  Future<void> unlinkFromProvider(String app, String provider) {
+    return channel.invokeMethod<void>(
+      'unlinkFromProvider',
+      <String, String>{'provider': provider, 'app': app},
+    );
+  }
+
+  @override
+  Future<void> updateEmail(String app, String email) {
+    return channel.invokeMethod<void>(
+      'updateEmail',
+      <String, String>{'email': email, 'app': app},
+    );
+  }
+
+  @override
+  Future<void> updatePhoneNumber(
```

Comment:
Make this method the same name as the one you're invoking? -> `updatePhoneNumberCredential`

Reply:
Done.